src/irclog2html/tests/test_irclogsearch.py

```diff
@@ -218,6 +222,128 @@ def doctest_print_search_results():
     """
 
 
+def doctest_search_page():
+    """Test search_page
+    Let's mock the dependency functions:
+
+        >>> patcher = mock.patch.multiple("irclog2html.irclogsearch",
+        ...                               print_search_form=mock.DEFAULT,
+        ...                               print_search_results=mock.DEFAULT)
+
+        >>> values = patcher.start()
+        >>> values['print_search_results'].return_value = "Formatter"
+
+    When there is a 'q' param in the GET query, the logs are searched:
+
+        >>> form = cgi.FieldStorage(
+        ...     environ={'QUERY_STRING': 'q=123', 'HTTP_METHOD': 'GET'})
+        >>> search_page("The stream", form, "/logs", "#dev*.logs")
+        'Formatter'
+        >>> values['print_search_results'].assert_called_once_with(
+        ...     '123', logfile_pattern='#dev*.logs',
+        ...     stream='The stream', where='/logs')
+
+    When there is no query, the search form is displayed:
+
+        >>> form = cgi.FieldStorage(environ={
+        ...     'HTTP_METHOD': 'GET', 'QUERY_STRING': ''})
+        >>> search_page("The stream", form, "/logs", "#dev*.logs")
+        >>> values['print_search_form'].assert_called_once_with('The stream')
+
+    Clean up:
+
+        >>> patcher.stop()
+
+    """
+
+
+def doctest_get_path():
+    """Test for get_path.
+
+    This function decides whether to search or to display a file based
+    on URL path:
+
+        >>> get_path(dict(PATH_INFO='/search'))
+        'search'
+
+        >>> get_path(dict(PATH_INFO='/#channel-2015-05-05.log.html'))
+        '#channel-2015-05-05.log.html'
```

Comment:
Oh, is the path info already URL-decoded?

I assume you've tested this manually, right?